### DISCUSSION TITLE
May I ask if it supports lower version Python=3.6

### DISCUSSION THREAD

**Discussion by zl530380** [utelle/apsw-sqlite3mc#11] [Q&A]
May I ask if it supports lower version Python=3.6

**Comment by rogerbinns** [comment]
Python 3.6 is not supported.  The current version of APSW uses CPython APIs that are not present in earlier versions.

You can find out which APSW version last supported a particular Python version [here](https://rogerbinns.github.io/apsw/about.html#python-versions).

You will be able to use the older APSW version with current sqlite3 multiple ciphers, but will have to arrange the compilation yourself.

**Comment by utelle** [comment]
> May I ask if it supports lower version Python=3.6

_apsw-sqlite3mc_ is based on Roger Binns' [_apsw_](https://github.com/rogerbinns/apsw/). As described in the [apsw documentation](https://rogerbinns.github.io/apsw/about.html#python-versions) _apsw_ supports [all supported Python versions](https://devguide.python.org/versions/). _Python **3.6**_ reached its end of life at the end of the year 2021 - that is, 3 years ago, and is therefore _**unsupported**_.

I would recommend you to upgrade to a more recent Python version. If that isn't an option for you, you could install [apsw 3.43.0.0](https://pypi.org/project/apsw/3.43.0.0/) - the last one that supports _Python 3.6_ -, compile _SQLite3 Multiple Ciphers_ yourself (or use pre-compiled binaries if you work under Windows), and exchange the _SQLite_ binary by the _SQLite3 Multiple Ciphers_  binary.

**Comment by utelle** [comment]
Since the question is not related to any real issue, I convert it to a discussion.
